File: dev/golang/sandbox/simple_http_server.go
package main

import (
	"io"
	"net/http"
)

func main() {
	http.HandleFunc("/hello", HelloHandler)
	http.HandleFunc("/ping", PingHandler)
	http.ListenAndServe(":8080", nil)
}

func HelloHandler(w http.ResponseWriter, req *http.Request) {
	io.WriteString(w, "Hello, world!\n")
}

func PingHandler(w http.ResponseWriter, req *http.Request) {
	io.WriteString(w, "Pong!\n")
}

File: dev/ruby/sandbox/simple_http_server.rb
require 'webrick'

server = WEBrick::HTTPServer.new(
  DocumentRoot: './',
  BindAddress: '127.0.0.1',
  Port: 8000
)

server.mount_proc '/hello' do |req, res|
  res.body = 'Hello, World!'
end

server.mount_proc '/ping' do |req, res|
  res.body = 'pong!'
end

trap("INT"){ server.shutdown }

server.start
